### PR TITLE
CLOUDP-251510: test/int: execute updates with conflict resolution

### PR DESCRIPTION
**Pull request by s-urbaniak**
Currently, deployment integration tests do not consider the fact the updates towards the Kubernetes API can error out caused by conflicting updates. The result is flaky behavior and can cause the following error message to appear in e2e logs:

```
2024-05-22T18:10:17.4967008Z       <*errors.StatusError | 0xc00041d0e0>: 
2024-05-22T18:10:17.4968127Z       Operation cannot be fulfilled on atlasdeployments.atlas.mongodb.com "test-deployment-aws-k8s": the object has been modified; please apply your changes to the latest version and try again
2024-05-22T18:10:17.4968224Z       {
2024-05-22T18:10:17.4968369Z           ErrStatus: {
2024-05-22T18:10:17.4968634Z               TypeMeta: {Kind: "", APIVersion: ""},
2024-05-22T18:10:17.4968790Z               ListMeta: {
2024-05-22T18:10:17.4968969Z                   SelfLink: "",
2024-05-22T18:10:17.4969196Z                   ResourceVersion: "",
2024-05-22T18:10:17.4969375Z                   Continue: "",
2024-05-22T18:10:17.4969618Z                   RemainingItemCount: nil,
2024-05-22T18:10:17.4969737Z               },
2024-05-22T18:10:17.4969919Z               Status: "Failure",
2024-05-22T18:10:17.4971538Z               Message: "Operation cannot be fulfilled on atlasdeployments.atlas.mongodb.com \"test-deployment-aws-k8s\": the object has been modified; please apply your changes to the latest version and try again",
2024-05-22T18:10:17.4971732Z               Reason: "Conflict",
2024-05-22T18:10:17.4971877Z               Details: {
2024-05-22T18:10:17.4972214Z                   Name: "test-deployment-aws-k8s",
2024-05-22T18:10:17.4972481Z                   Group: "atlas.mongodb.com",
2024-05-22T18:10:17.4972728Z                   Kind: "atlasdeployments",
2024-05-22T18:10:17.4972891Z                   UID: "",
2024-05-22T18:10:17.4973065Z                   Causes: nil,
2024-05-22T18:10:17.4973292Z                   RetryAfterSeconds: 0,
2024-05-22T18:10:17.4973409Z               },
2024-05-22T18:10:17.4973558Z               Code: 409,
2024-05-22T18:10:17.4973662Z           },
2024-05-22T18:10:17.4973785Z       }�[0m
2024-05-22T18:10:17.4974735Z   �[38;5;9mIn �[1m[It]�[0m�[38;5;9m at: �[1m/home/runner/work/mongodb-atlas-kubernetes/mongodb-atlas-kubernetes/test/int/deployment_test.go:210�[0m �[38;5;243m@ 05/22/24 18:08:24.802�[0m
```

This fixes it

### All Submissions:

* [ ] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?
* [ ] Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if there is one).
* [ ] Update docs/release-notes/release-notes-template.md if your changes should be included in the release notes for the next release.

### CODE REVIEW

File: test/int/deployment_test.go [hunk 1]
```diff
@@ -1810,3 +1863,28 @@ func mergedAdvancedDeployment(
...
+
+	clientObj := any(obj).(api.AtlasCustomResource)
+	Eventually(func(g Gomega) bool {
+		return resources.CheckCondition(k8sClient, clientObj, api.TrueCondition(api.ReadyType), validateDeploymentUpdatingFunc(g))
+	}).WithTimeout(timeout).WithPolling(interval).Should(BeTrue())
+
+	return obj
+}
+
+func retryUpdateOnConflict[T any](ctx context.Context, k8s client.Client, key client.ObjectKey, mutator func(*T)) (*T, error) {
```
Comment (helderjs) [inline test/int/deployment_test.go:1879]: I have the feeling I have seen this function before. Also, it seems to be generic enough to be used in many cases. Shouldn't it be moved to the help utility package?

**Review by helderjs — APPROVED** [review]
LGTM! Just left a suggestion. Also would run cloud tests to see deployment goes green with no issue

File: test/int/deployment_test.go (hunk 1 above)
Comment (s-urbaniak) [reply test/int/deployment_test.go:1879]: I would do this on demand 👍

Comment (s-urbaniak) [reply test/int/deployment_test.go:1879]: https://github.com/mongodb/mongodb-atlas-kubernetes/pull/1625 here we go :-)

**Pull request merged**